docs/proposals/002-api-proposal/proposal.md

```diff
@@ -0,0 +1,256 @@
+- SLO objectives for the UseCase
+- The Pools this UseCase is relevant to 
+
+### Spec
+
+**ModelGroup**
+```golang
+// ModelGroup represents a set of LLM use cases that are multiplexed onto one 
+// or more backend pools. This is generally owned by the "LLM Use Case Owner"
+// persona, which can be teams in an organization. Plural ModelUseCases are
+// allowed as a configuration convenience to the user. Allowing a user who
+// has multiple usecases across multiple pools (with the same config) to
+// specify the configuration exactly once, and deploy to many pools 
+// simultaneously, allowing for a simple config and single source of truth
+// for a given user. ModelUseCase names are unique for a given BackendPool,
+// if the name is reused, an error will be  shown on the status of a 
+// ModelGroup that attempted to reuse. 
+type ModelGroup struct {
+        metav1.ObjectMeta
+        metav1.TypeMeta
+
+        Spec ModelGroupSpec
+}
+
+type ModelGroupSpec struct {
+        // Defines the use cases in the set.
+        // UseCases can be in 2 priority classes, CRITICAL and NONCRITICAL. 
+        // Priority class is implicit, and by specifying an Objective,
+        // places the UseCase in the CRITICAL priority class.
+        UseCases   []ModelUseCases
+        // Reference to the backend pools that the use cases registers to.
+        PoolRef []corev1.ObjectReference
+}
+
+// ModelGroup defines the policies for routing the traffic of a use case, this includes performance objectives 
+// and traffic splitting between different versions of the model.
+type ModelUseCases struct {
+        // The name of the model as the users set in the "model" parameter in the requests.
+        // The model name should be unique among the use cases that reference the same backend pool.
+        // This is the parameter that will be used to match the request with. In the future, we may
+        // allow to match on other request parameters. The other approach to support matching on 
+        // on other request parameters is to use a different ModelName f HTTPFilter 
+        ModelName string
+        // Optional
+        // Use cases with an objective have higher priority than use cases without.
+        // IMPORTANT: By specifying an objective, this places the UseCase in a higher priority class than UseCases without a defined priority class. 
+        // In the face of resource-scarcity. Higher priority requests will be preserved, and lower priority class requests will be rejected.
+        Objective *Objective
+        // Optional.
+	// Allow multiple versions of a model for traffic splitting. 
+	// If not specified, the target model name is defaulted to the modelName parameter.
+        TargetModels []common.TargetModel
+}
+
+
+
+// TargetModel represents a deployed model or a LoRA adapter.
+type TargetModel struct {
+        // The name of the adapter as expected by the ModelServer.
```

Comment:
You probably need a few more sentences in this description.  Explain that model servers are expected to be configured for access to a set of LoRA adapters and that the naming of the adapter is defined by the inference platform admin who deployed the model servers.  Explain what happens if the adapter doesn't match a value (probably the model server returns a specific type of error, but explain what *should* happen according to our "model server protocol" for vLLM).

Reply:
Revised